src/vs/workbench/services/textmodelResolver/common/textModelResolverService.ts

```diff
@@ -77,7 +78,8 @@ class ResourceModelCollection extends ReferenceCollection<Promise<IResolvedTextE
 		if (this.providers.has(resource.scheme)) {
 			await this.resolveTextModelContent(key);
 
-			const model = this.instantiationService.createInstance(TextResourceEditorModel, resource);
+			const cls = resource.scheme === 'vscode-notebook-cell' ? NotebookCellResourceEditorModel : TextResourceEditorModel;
+			const model = this.instantiationService.createInstance(cls, resource);
```

Comment:
Hack, all cell text models are in virtual documents, hence readonly by deafult, hence created `NotebookCelLResourceModel`